filebeat/docs/fields.asciidoc

```diff
@@ -800,6 +800,42 @@ elasticsearch Module
 
 
 
+[float]
+== gc fields
+
+
+
+
+*`elasticsearch.gc.relative_process_timestamp_secs`*::
++
+--
+type: float
+
+Garbage collection threads total stop time seconds.
```

Comment:
@inqueue Just one clarification on the meaning of those: The metrics that are captured here are not necessarily tied to garbage collection. These times are actually related to safepoints. At a safepoint all application threads are halted and the JVM can do any administrative work that requires that the application cannot interfere. That might include such things as deoptimization, biased locking or garbage collection. 

The first metric tells for how long the application threads have been halted (the duration of the pause due to the safepoint) and the second one tells how long it took from the point in time the JVM requested a safepoint until all application threads could been halted.

Reply:
Thanks @danielmitterdorfer, that makes sense. I did want to capture that metric in this initial commit as it is probably the most frequently logged in gc.log. Can we work with you to isolate the most meaningful metrics? We also intend to incorporate GC metrics captured in the Kafka Filebeat module- https://www.elastic.co/blog/monitoring-kafka-with-elastic-stack-1-filebeat.

Reply:
> Can we work with you to isolate the most meaningful metrics?

Yes. The most meaningful ones in my opinion are:

* Absolute timestamp of an event
* Type of pause (young GC, old GC)
* Pause time (how long was the application stopped?), GC (phase) duration (might happen concurrently)
* Amount of memory cleaned up

Although I think it will get very tricky to parse all those variations:

* The log file format has changed with JDK 9 (so called "unified logging").
* There are different garbage collection algorithms (e.g. CMS, G1)

https://www.youtube.com/watch?v=M9AYK5PtX40 should give a good idea about this.

Reply:
You're welcome. Glad I could help.